presto-hive/src/main/java/com/facebook/presto/hive/HiveTypeTranslator.java

```diff
@@ -152,7 +152,62 @@ else if (varcharLength == VarcharType.UNBOUNDED_LENGTH) {
                             .collect(toList()));
         }
         return defaultHiveType
-                .orElseThrow(() -> new PrestoException(NOT_SUPPORTED, format("Unsupported Hive type: %s", type)))
+                .orElseThrow(() -> new PrestoException(NOT_SUPPORTED, format("No default Hive type provided for unsupported Hive type: %s", type)))
                 .getTypeInfo();
     }
+
+    public static boolean isSupportedHiveType(Type type)
```

Comment:
Heh in some ways this duplication of logic with `translate` is also unfortunate.

I like pushing the nastiness into this utility function, but @wenleix  would you rather not duplicate it and keep the exception handling?

Reply:
Hopefully PAGEFILE will be used as default value for temporary file format, then we can turn off this session property and bypass this function.

Reply:
> I like pushing the nastiness into this utility function, but @wenleix would you rather not duplicate it and keep the exception handling?

@aweisberg : I agree this is not ideal. But in general we don't want exception handling being used for normal (or expected) business logic. Exception handling code path should be unexpected situation.

We could refactor `translate` to return an `Optional`, for example, and returns `Optional.emtpy()` to indicate there is no viable translation (and throw exception in callee). But I am OK to just keep the current code and hopefully we can remove them once NATIVE FORMAT is 100% rolled out (or, we refactor it).